### PR TITLE
Correctly account for date timezone offsets

### PR DESCRIPTION
**Pull request by jamie--stewart**
Patch Diem functionality such that

`1st April 2020 00:00:00`, which is in British Summer Time, correctly outputs a `Diem` with `2020-04-01` as its date.

### CODE REVIEW

File: src/diem.test.ts [hunk 1]
```diff
@@ -30,6 +31,15 @@ inAllTimezonesIt.each = <F extends AnyFn>(params: Array<Parameters<F>>) => (desc
 
 describe('Diem', () => {
 	describe('test construction', () => {
+
+		it('should handle midnight in BST', () => {
```
Comment (tdalziel) [inline src/diem.test.ts:35]: Good catch with this PR. This edge case (well, it literally happens twice a year so maybe not) has caught me so many times.
